llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -715,8 +715,16 @@ MCSymbol *AsmPrinter::getSymbolPreferLocal(const GlobalValue &GV) const {
   return TM.getSymbol(&GV);
 }
 
-/// EmitGlobalVariable - Emit the specified global variable to the .s file.
 void AsmPrinter::emitGlobalVariable(const GlobalVariable *GV) {
+  MaybeAlign RequiredAlignment = getRequiredGlobalAlignment(*GV);
```

Comment:
Should we also do this in `AsmPrinter::emitGlobalConstant`?

Reply:
For MTE, we don't really need this. We can add this in a followup if that's required by CHERI.

Reply:
For CHERI it is only needed if a reference to a large global constant is passed somewhere else, since for local accesses the compiler can elide bounds. Adding it when needed sounds good to me.